services/dummy-test-service/main.go

```diff
@@ -1,7 +1,295 @@
+			for k := 0; k < len(Data["Users"][i].City[j].Buildings); k++ {
+				if (username == Data["Users"][i].Username) && (cityid == Data["Users"][i].City[j].CityID) &&
+					(buildingid == Data["Users"][i].City[j].Buildings[k].BuildingID) {
+					building_aux = Data["Users"][i].City[j].Buildings[k]
+				}
+			}
+		}
+	}
+	return building_aux
+}*/
+
+func getRoot(w http.ResponseWriter, r *http.Request) {
+	fmt.Printf("got / request\n")
+	io.WriteString(w, "Welcome to Stickian!\n")
+}
+
+func (s *serverHandler) handleDB(w http.ResponseWriter, r *http.Request) {
+
+	w.WriteHeader(http.StatusCreated)
+	w.Header().Set("Content-Type", "application/json")
+
+	jsonResp, err := json.Marshal(s.db.getDB())
+	if err != nil {
+		log.Fatalf("Error happened in JSON marshal. Err: %s", err)
+	}
+	w.Write(jsonResp)
+	//return
+}
+
+func (s *serverHandler) handleUser(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+
+	w.WriteHeader(http.StatusCreated)
+	w.Header().Set("Content-Type", "application/json")
+
+	jsonResp, err := json.Marshal(s.db.getUser(vars["username"]))
+	if err != nil {
+		log.Fatalf("Error happened in JSON marshal. Err: %s", err)
+	}
+	w.Write(jsonResp)
+	//return
+}
+
+func (s *serverHandler) handleCity(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+
+	w.WriteHeader(http.StatusCreated)
+	w.Header().Set("Content-Type", "application/json")
+
+	var_aux1, err := strconv.Atoi(vars["cityid"])
+	if err != nil {
+		fmt.Printf("Error converting String to Int\n")
+		panic(err)
+	}
+
+	jsonResp, err := json.Marshal(s.db.getCity(vars["username"], var_aux1))
+	if err != nil {
+		log.Fatalf("Error happened in JSON marshal. Err: %s", err)
+	}
+	w.Write(jsonResp)
+	//return
+}
+
+func (s *serverHandler) handleBuilding(w http.ResponseWriter, r *http.Request) {
+
+	vars := mux.Vars(r)
+
+	w.WriteHeader(http.StatusCreated)
+	w.Header().Set("Content-Type", "application/json")
+
+	var_aux := vars["cityid"]
+
+	var_aux1, err := strconv.Atoi(var_aux)
+	if err != nil {
+		fmt.Printf("Error converting String to Int\n")
+		panic(err)
+	}
+
+	var_aux2 := vars["buildingid"]
+
+	var_aux3, err := strconv.Atoi(var_aux2)
+	if err != nil {
+		fmt.Printf("Error converting String to Int\n")
+		panic(err)
+	}
+
+	jsonResp, err := json.Marshal(s.db.getBuilding(vars["username"], var_aux1, var_aux3))
+	if err != nil {
+		log.Fatalf("Error happened in JSON marshal. Err: %s", err)
+	}
+	w.Write(jsonResp)
+	//return
+}
+
+func (s *serverHandler) init_db() mockDB {
+
+	var database mockDB
+	database.dbmock = s.db.getDB()
+
+	return database
+}
 
 func main() {
-	fmt.Printf("hello world")
+
+	server := serverHandler{
+		db: &mockDB{},
+	}
+
+	//user_test := "MCKarol123"
+	//city_test := "1233"
+	//building_test := "1"
+
+	server.init_db()
+
+	client, ctx, cancel, err1 := connect("mongodb://localhost:27017")
+	if err1 != nil {
+		panic(err1)
+	}
+
+	ping(client, ctx)
+
+	r := mux.NewRouter()
+
+	r.HandleFunc("/", getRoot)
```

Comment:
Get root can be the "catch all" and then you remove the path in the handle function (and re-route to smaller handler functions)